### DISCUSSION TITLE
Versioning

### DISCUSSION THREAD

**Discussion by DE-cr** [fredlcore/BSB-LAN#497] [Q&A] [answered]
Just curious: What kind of changes lead to a new version number for BSB-LAN? (I've noticed that there's been plenty of changes in the software on GitHub that haven't lead to a new version number being applied.)

**Comment by fredlcore** [comment, marked as answer]
Only the versions of the release versions really count and these try to follow the standard versioning scheme.

**Reply by DE-cr** [reply]
My bsb-lan unit tells me there's a new version 3.0.3, but on github the latest release is given as v2.2. How do these numbers correlate?
...and what's the changes in 3.0.3? The changelog in BSB_LAN.ino doesn't mention this version.

**Reply by 1coderookie** [reply]
See notes on updates, chap. 2.1.3:
![Bildschirmfoto von 2022-11-19 10-59-26](https://user-images.githubusercontent.com/25869362/202845445-b0fc9359-b7c4-484a-999d-c2f43cd2c9ad.png)
![Bildschirmfoto von 2022-11-19 11-02-15](https://user-images.githubusercontent.com/25869362/202845495-0a769298-0ed7-43a3-9ef1-8ef14d8984d8.png)

Then in chap 3.3 the note about v2.2:
![Bildschirmfoto von 2022-11-19 11-02-53](https://user-images.githubusercontent.com/25869362/202845509-b1c62566-88f1-42d5-9a2f-8ba21ac23131.png)

Summary v2.2 vs v3.x: Besides bugfixes and stuff, v3.x doesn't come with a 'comprehensive' custom_defs.h anymore, you _need_ to get your device specific custom_defs.h to get full access to your controller.

**Reply by DE-cr** [reply]
Understood.
So the changes from 3.0.1 => 3.0.2 seem to be bugfixes "only", and nothing worth mentioning in the changelog in BSB_LAN.ino. Hmm.

**Comment by DE-cr** [comment]
Ok, what I've meant to ask is: When do you call it a new release version?

**Comment by fredlcore** [comment]
Whenever I feel that some major works have been completed or are going to happen.
